Add Python Turtle workshop (#1340)
hackclub/hackclub: workshops/python_turtle/README.md

---
name: 'Python Turtle'
description: 'Exciting Visuals with Python Turtle'
author: '@YashKalbande'
---

# Exciting Visuals with Python Turtle

## Introduction

> Welcome! my friends.  My name is Yash, and I am a technology enthusiast. Have you ever heard that you can create exciting visuals with help of Python?  My goal is to help you understand the python turtle library. With the help of the turtle library, we can create stunning visuals. First, I'm going to present to you the basics of the python turtle library. Then I'll share with you shapes like Triangle, Star, Hexagon. Finally, I'll ask you to create some more visual using ideas given in the end.

Reviewer: Don’t introduce yourself—this should stand alone as a tutorial

Author: done.

![Benzen ring with Turtle](https://cloud-36kk4uh0h.vercel.app/turtle_benzene_ring.gif)

## Quickstart

> TODO - Explain how to install and setup Python 3 on [repl.it](https://repl.it/languages/python3) for turtle graphics.

Reviewer: There shouldn’t be TODO

Reviewer 2: fyi @YashKalbande, you don't need to install python3 it's preinstalled :)

Author: done

![Square](https://cloud-36kk4uh0h.vercel.app/square.gif)

The instructions in your program tell the "turtle" how to move. The turtle draws a line behind it as it moves. This program draws a square. The steps given to the program are:

1. Move forward 100 steps. (In the beginning, the turtle is facing to the right.)
2. Turn 90 degrees to the left.
3. Move forward 100 steps.
4. Turn 90 degrees to the left.
5. Move forward 100 steps.
6. Turn 90 degrees to the left.
7. Move forward 100 steps. The turtle has ended up where it started.

With these seven steps, the turtle draws a square. The `from turtle import *` is an instruction needed at the beginning of all of your turtle programs. It imports the turtle module so you can do the turtle instructions.

There are many instructions like left() and forward(). These instructions are called functions. This tutorial explains many of the functions in the turtle module. When you learn more about these functions, you will be able to draw many different shapes and beautiful pictures!

There are many instructions like left() and forward(). These instructions are called functions. Commonly used turtle methods are :

> | Method        | Parameter           | DESCRIPTION  | Example |
> | ------------- |:-------------:| -----:| --------:|

Reviewer: ```suggestion
| Method        | Parameter           | Description  | Example |
| ------------- | ------------- | ----- | -------- |
```

```python
from turtle import *

fillcolor('purple')
pensize(10)
pencolor('black')
forward(100)

begin_fill()
forward(100)
left(90)
forward(100)
left(90)
forward(100)
left(90)
forward(100)
left(90)
end_fill()
```

### fillcolor(), fillcolor(*color*), fillcolor((*red*, *green*, *blue*)), fillcolor(*red*, *green*, *blue*)

The fillcolor() function sets the color of the filled-in shape when end_fill() is called. The fillcolor() function can be passed a string of the color, such as 'red' or 'black'. Or, the fillcolor() function can be passed an "RGB color tuple" (see the [Color](#color) section).

### begin_fill()

The begin_fill() starts recording the moves that will be the outline of the filled-in shape. The filled-in shape will not be drawn until end_fill() is called.

### end_fill()

The end_fill() function will stop recording the moves for the filled-in shape and draw the shape.

> To get more information about color [visit hear](https://docs.python.org/3/library/turtle.html#turtle.color)

Reviewer: ```suggestion
To get more information about color, [visit here](https://docs.python.org/3/library/turtle.html#turtle.color).
```

## Hacking

> - There are endless ways to make this turtle completely your own! Go back through your code and look for anything that can be played with. Try changing the default values of your forward, angle, color, size, etc.
> 
> - Here are some examples to give you some more idea:

Reviewer: ```suggestion

There are endless ways to make this turtle completely your own! Go back through your code and look for anything that can be played with. Try changing the default values of your forward, angle, color, size, etc.

Here are some examples to give you some ideas:
```
